GitUtils.ps1

```diff
@@ -349,10 +372,21 @@ function Add-SshKey() {
             }
         }
     } else {
-        $sshAdd = Get-Command ssh-add -TotalCount 1 -ErrorAction SilentlyContinue
-        $sshAdd = if ($sshAdd) {$sshAdd} else {Find-Ssh('ssh-add')}
+        $agentPid = (Get-Item Env:SSH_AGENT_PID -ErrorAction SilentlyContinue).Value
+        $sshAdd = '';
+        if($agentPid){
+            $agentPath = (Get-Process -Id $agentPid -FileVersionInfo -ErrorAction SilentlyContinue).Filename
+            if($agentPath){
+                $sshPath = Split-Path ($agentPath) -Parent
+            }
+            if($sshPath){
+                $sshAdd = Get-Command (Join-Path $sshPath 'ssh-add')
+            }
+        }
+        if(!$sshAdd){
+            $sshAdd = Find-Ssh('ssh-add')
+        }

```

Comment:
If you look carefully, $agentPid is set [whether it's openssh or git ssh], so the new block of code always gets executed and returns right ssh-add. the old code is unnecessary Find-Ssh. I just kept it for the sake of not breaking anyone in the worst case 😄